### PR TITLE
Topbar.js: refactoring most pressing things.

### PR DESCRIPTION
**Pull request by Gnito**
Lots of refactoring - there's still room to improve, but I guess that can be done later.

### CODE REVIEW

File: client/app/components/sections/Topbar/Topbar.js [hunk 1]
```diff
@@ -22,14 +22,27 @@ import MenuMobile from '../../composites/MenuMobile/MenuMobile';
...
+
+const LABEL_TYPE_MENU = 'menu';
+const LABEL_TYPE_DROPDOWN = 'dropdown';
+const SEARCH_PARAMS_TO_KEEP = ['view', 'locale'];
+const SEARCH_PARAMS = ['q', 'lq'];
+const DEFAULT_CONTEXT = {
+  marketplace_color1: styleVariables['--customColorFallback'],
+  marketplace_color2: styleVariables['--customColor2Fallback'],
+  loggedInUsername: null,
+};
```
Comment (mporkola) [inline client/app/components/sections/Topbar/Topbar.js]: 👍 for collecting these up here

**Comment by mporkola** [comment]
Good refactoring this far, maybe change the PR name :)

File: client/app/components/sections/Topbar/Topbar.js [hunk 2]
```diff
@@ -197,12 +173,43 @@ class Topbar extends Component {
...
+      }) :
+      {};
 
-    const pathParams = { return_to: location };
-    const loginRoute = this.props.routes.login_path ? this.props.routes.login_path(pathParams) : '#';
-    const signupRoute = this.props.routes.sign_up_path ? this.props.routes.sign_up_path() : '#';
-
-    const mobileMenuProps = this.props.menu ?
+    // mobile menu props
+    const isAdmin = !!(this.props.user && this.props.user.isAdmin && loggedInUsername);
```
Comment (mporkola) [inline client/app/components/sections/Topbar/Topbar.js]: `isAdmin` is also used in `ProfileDropdown`, so maybe not under "mobile menu props"

File: client/app/components/sections/Topbar/Topbar.js [hunk 3]
```diff
@@ -197,12 +173,43 @@ class Topbar extends Component {
       }) :
       null;
 
+    // menu props
+    const hasMenuContext = !!this.props.menu;
```
Comment (mporkola) [inline client/app/components/sections/Topbar/Topbar.js]: I don't believe you meant "context"? `hasMenuContent`, or `hasMenu`?

**Comment by mporkola** [comment]
Good stuff, minor comments

File: client/app/components/sections/Topbar/Topbar.js (hunk 3 above)
Comment (Gnito) [inline client/app/components/sections/Topbar/Topbar.js]: hasMenuProps should be clear?

File: client/app/components/sections/Topbar/Topbar.js [hunk 4]
```diff
@@ -22,14 +22,27 @@ import MenuMobile from '../../composites/MenuMobile/MenuMobile';
...
 
-const profileDropdownActions = function profileDropdownActions(routes, username) {
+
+const LABEL_TYPE_MENU = 'menu';
+const LABEL_TYPE_DROPDOWN = 'dropdown';
+const SEARCH_PARAMS_TO_KEEP = ['view', 'locale'];
+const SEARCH_PARAMS = ['q', 'lq'];
+const DEFAULT_CONTEXT = {
+  marketplace_color1: styleVariables['--customColorFallback'],
+  marketplace_color2: styleVariables['--customColor2Fallback'],
```
Comment (sktoiva) [inline client/app/components/sections/Topbar/Topbar.js]: Wasn't the customColor2 removed in the new layout? It shouldn't affect anything anymore? The only place where it was used was the Post a new listing button color and even the option to configure it is removed if the new topbar is enabled.

Comment (Gnito) [inline client/app/components/sections/Topbar/Topbar.js]: From current version, we might use it later though.
But I'll remove it from topbar.js related code.

**Pull request merged**